docs/experiments-doc2query.md

```diff
@@ -87,5 +87,60 @@ So, this simple trick improves MRR by a bit over baseline Doc2query.
 
 ## TREC CAR
 
-Instructions coming soon!
+We will now describe how to reproduce the TREC-CAR results of our model BM25+doc2query presented in the paper.
 
+To start, download the TREC-CAR dataset and the predicted queries (i.e., document expansions):
+```
+mkdir trec_car
+
+wget http://trec-car.cs.unh.edu/datareleases/v2.0/paragraphCorpus.v2.0.tar.xz -P trec_car
+wget https://storage.googleapis.com/neuralresearcher_data/doc2query/data/aligned5/pred-test_topk10.tar.gz -P trec_car
+
+tar -xf trec_car/paragraphCorpus.v2.0.tar.xz -C trec_car
+tar -xf trec_car/pred-test_topk10.tar.gz -C trec_car
+```
+
+To confirm, `paragraphCorpus.v2.0.tar.xz` should have an MD5 checksum of `a404e9256d763ddcacc3da1e34de466a` and
+ `pred-test_topk10.tar.gz` should have an MD5 checksum of `b9f98b55e6260c64e830b34d80a7afd7`.
+
+These are the predicted queries based on our seq2seq model, based on top _k_ sampling with 10 samples for each document in the corpus.
+There are as many lines in the above file as there are documents; all 10 predicted queries are concatenated on a single line.
+
+Now let's create a new document collection by concatenating the predicted queries to the original documents:
+
+```
+python src/main/python/treccar/augment_collection_with_predictions.py \
+  --collection_path trec_car/paragraphCorpus/dedup.articles-paragraphs.cbor --output_folder trec_car/collection_jsonl_expanded_topk10 \
+  --predictions trec_car/pred-test_topk10.txt --stride 1
+```
+
+This augmentation process might take 2-3 hours.
+
+We can then index the expanded documents:
+
+```
+nohup sh target/appassembler/bin/IndexCollection -collection JsonCollection \
+-generator LuceneDocumentGenerator -threads 40 -input trec_car/collection_jsonl_expanded_topk10 -index \
+trec_car/lucene-index.car17v2.0
+```
+
+And retrieve the test queries:
+
+```
+sh target/appassembler/bin/SearchCollection -topicreader Car -index trec_car/lucene-index.car17v2.0 \
+-topics src/main/resources/topics-and-qrels/topics.car17v2.0.benchmarkY1test.txt \
+-output trec_car/run.car17v2.0.bm25.topics.car17v2.0.benchmarkY1test.txt -bm25
+```
+
+Evaluation is performed with `trec_eval`:
+```
+eval/trec_eval.9.0.4/trec_eval -c -m map -c -m recip_rank \
+src/main/resources/topics-and-qrels/qrels.car17v2.0.benchmarkY1test.txt \
+trec_car/run.car17v2.0.bm25.topics.car17v2.0.benchmarkY1test.txt
+```
+
+With the above commands, you should be able to replicate the following results:
+```
+map                   	all	0.1807
```

Comment:
https://arxiv.org/abs/1904.08375 reports AP of 17.8... at least note that it's slightly different.
Any idea why? Is just from Lucene8 upgrade?

Reply:
It is probably because we v1.5 corpus in the arXiv paper vs v2.0 in the EMNLP submission.

Reply:
I added a note.